frontend/packages/process-editor/src/hooks/useBpmnEditor.ts

```diff
@@ -48,14 +80,18 @@ export const useBpmnEditor = (): UseBpmnViewerResult => {
         await modelerInstance.importXML(bpmnXml);
         const canvas: any = modelerInstance.get('canvas');
         canvas.zoom('fit-viewport');
+        // Reset the dataTasks tracking states when the editor is initialized
+        setDataTasksAdded([]);
+        setDataTasksRemoved([]);
       } catch (exception) {
         console.log('An error occurred while rendering the viewer:', exception);
       }
     };
 
     initializeEditor();
     initializeUnsavedChangesCount();
-  }, [bpmnXml, modelerRef, setBpmnDetails, setNumberOfUnsavedChanges]);
+    initializeChangesStatus();
+  }, [bpmnXml, modelerRef, setBpmnDetails, setNumberOfUnsavedChanges, setDataTasksAdded]);
```

Comment:
I have re-structured this a bit, as I also noticed a different bug (that has been there a long time) relating to initializing the different event listeners. 
Don't think we'll be able to avoid the warning entirely, as including `getModeler` really causes big issues, and we have not been able to fix it in a good way.